pkg/preparer/orchestrate.go

```diff
@@ -203,6 +219,11 @@ func (p *Preparer) verifySignature(manifest pods.PodManifest, logger logging.Log
 func (p *Preparer) installAndLaunchPod(newManifest *pods.PodManifest, pod Pod, logger logging.Logger) bool {
 	// do not remove the logger argument, it's not the same as p.Logger
 
+	if p.podIdForbidden(newManifest.ID()) {
```

Comment:
One other small code etiquette thing because this function is now >100 lines :oncoming_police_car: :ambulance: lets move this into a separate function we invoke.